lib/solidus_importer/base_importer.rb

```diff
@@ -2,8 +2,17 @@
 
 module SolidusImporter
   class BaseImporter
+    attr_reader :options, :logger
+
     def initialize(options)
       @options = options
+      @logger ||= begin
+        if @options[:logger] && !@options[:logger].is_a?(Class)
+          @options[:logger]
+        else
+          (@options[:logger] || ::SolidusImporter::BaseLogger).new
+        end
+      end
```

Comment:
Would also be good to give the logger a chance to interact with the context itself, so we might decide to pass it to `.new` or even go a step further and always use a callable that takes in the importer as a parameter, e.g.: 

```suggestion
      @logger = @options[:logger] || ->(context) { ::SolidusImporter::BaseLogger.new(context) }
```

and then prepare it in https://github.com/nebulab/solidus_importer/blob/cf0584b779eebce6b7fc675ccbe236e767873724/lib/solidus_importer/process_row.rb#L15

That means that we can initialize the importer with anything that can respond to logger methods and ignore the _context_ parameter if we want to.

E.g. passing a tagged Rails logger:

```rb
SolidusImporter::BaseImporter.new(
  …
  logger: ->(context) { Rails.logger.tagged(context[:row_id]) }
)
```